exwm-manage.el

```diff
@@ -405,22 +405,14 @@ manager is shutting down."
         (xcb:flush exwm--connection))
       (let ((kill-buffer-func
              (lambda (buffer)
-               (with-current-buffer buffer
-                 (let ((kill-buffer-query-functions nil)
-                       (floating exwm--floating-frame))
-                   (kill-buffer)
-                   (when floating
-                     (select-window
-                      (frame-selected-window exwm-workspace--current))))))))
-        (if (not (active-minibuffer-window))
-            ;; Kill the buffer as usual.
-            (funcall kill-buffer-func buffer)
-          ;; This can happen when this buffer was requested to be killed
-          ;; from the minibuffer (e.g. with `ido-kill-buffer-at-head').
-          ;; We have to exit the minibuffer first or there'll be a
-          ;; "selecting deleted buffer" error.
-          (run-with-idle-timer 0 nil kill-buffer-func buffer)
-          (exit-minibuffer))))))
+               (let ((kill-buffer-query-functions nil))
+                 (kill-buffer buffer)))))
+        (when exwm--floating-frame
+          (select-window
+           (frame-selected-window exwm-workspace--current)))
+        (when (active-minibuffer-window)
+          (exit-minibuffer))
```

Comment:
We can't call `exit-minibuffer` before other code I think.

Reply:
Here's the test I was using to reproduce the issue:
- Run Firefox
- Open a Firefox dialog box with `\C-o`
- `M-x` to open the mini-buffer
- Click the `Close` button in the Firefox dialog box

With the above sequence I get: `execute-extended-command: ‘’ is not a valid command name` in the minibuffer and I'm able to continue using Emacs by clicking one of the other buffer windows.

If I move `(exit-minibuffer)` to after `(run-with-idle-time ...)` then Emacs throws the `[XELB Error] Selecting deleted buffer` message and becomes unresponsive.

Open to suggestions.

Reply:
Ah, I just realized why `exit-buffer` can't be called - because it throws an exception and the buffer never gets deleted. Let me investigate further.